gdk4/src/event.rs

```diff
@@ -0,0 +1,527 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use crate::{
+    AxisUse, CrossingMode, Device, Display, Drop, EventType, KeyMatch, ModifierType, NotifyType,
+    ScrollDirection, Seat, Surface, TimeCoord, TouchpadGesturePhase,
+};
+use glib::translate::*;
+use glib::{StaticType, Type};
+use std::fmt;
+use std::mem;
+
+glib::wrapper! {
+    #[derive(Debug)]
+    pub struct Event(Shared<ffi::GdkEvent>);
+
+    match fn {
+        ref => |ptr| ffi::gdk_event_ref(ptr),
+        unref => |ptr| ffi::gdk_event_unref(ptr),
+    }
+}
+
+pub const NONE_EVENT: Option<&Event> = None;
+
+impl StaticType for Event {
+    fn static_type() -> Type {
+        unsafe { from_glib(ffi::gdk_event_get_type()) }
+    }
+}
+
+unsafe impl Send for Event {}
+unsafe impl Sync for Event {}
+
+impl Event {
+    /*
+    pub fn downcast<T: EventTypeTrait>(self) -> Result<T, Event> {
+        unsafe {
+            if self.get_type() == T::static_type() {
```

Comment:
I'm not sure how can I add a get_type method here ^^

Reply:
Do the different event types have different `GType`s?

I think you'll have to go with `gdk_event_get_event_type()` and `GdkEventType` here instead.

Reply:
I ended up adding a new trait that allows me to know which event types are supported by which event :) Seems to build at least